### DISCUSSION TITLE
Rumble support?

### DISCUSSION THREAD

**Discussion by cathery** [cathery/sys-con#1] [Q&A]
There is currently no implementation of rumble features in the app due to the fact that neither `AbstractedPad` nor `Hdls` functions in switch firmware allow you to get the vibration value of the device. 

You could try to work around this by registering a `VibrationDeviceHandle` for the controller you wish to use, but I can't seem to get it working properly or change any values. There is also the issue of no actual vibration being sent because of a bug that requires you to re-enable controller vibration in switch system settings.

If you know how to solve this issue, or have anything to add, please let me know.

**Comment by Ntemis** [comment]
Hi, how you explain the rumble support in Mayflash Magic NS and 8bitdo's https://www.8bitdo.com/wireless-usb-adapter/ ?

**Comment by cathery** [comment]
8bitdo fakes each controller as an official Nintendo controller, which is handled directly and internally by the firmware. This sysmodule relies on hiddbg's virtual controller functionality, which doesn't seem to receive any rumble input/output.

**Comment by Ntemis** [comment]
got you and thanks for the fix to have 4 controllers with one dongle.
Shame though, i hope to be supported in the future somehow in libnx

**Comment by SmartBoy84** [comment]
So this is not possible - Would really love this as it is the last feature stopping your program from being perfect! I'll happily donate if you want?

**Comment by Ntemis** [comment]
check this out
https://github.com/switchbrew/libnx/issues/347

**Comment by cathery** [comment]
Yeah I know. I already told you

**Comment by cathery** [comment]
As mentioned by yellows8 in the linked issue, it seems like the system doesn't support rumble for virtual controllers, which is what sys-con relies on.

**Reply by amit27592** [reply]
I believe ndeadly has added rumble support via mitm in the Bluetooth module ([here](https://github.com/ndeadly/MissionControl/commit/1f2d0dc6ba0ebb2a0498453dba70b974f2bf6f71)).
Any chance a similar approach can be done on sys-con as well?

**Reply by rom4ster** [reply]
I would love to see a USB MITM

**Comment by Ntemis** [comment]
@cathery ping on what @amit27592 reported.
Do we have any hope to see rumble support on this now?
Is the only driver that has Xbox360 dongle support so fixing rumble we can finally add 4 controllers to the switch is its a big deal to many users atm.
Thank you for your time
